module_7/csv_module.py

```diff
@@ -0,0 +1,100 @@
+                if word in counts:
+                    counts[word] += 1
+                else:
+                    counts[word] = 1
+
+        return counts
+
+    def remove_spaces(self, input_str):
+        """
+        Method to remove spaces and white characters from input string
+        :param input_str: string to be transformed
+        :return: transformed string
+        """
+        return re.sub(r"\s", "", input_str)
+
+    def analyze_count_letters(self, input_str):
+        """
+        Method that counts occurrence if letter, count letter in upper case, percentage
+        :param input_str: string to be analysed
+        :return: res - dictionary of letters and counts
+        """
+        res = {i.lower(): [input_str.lower().count(i.lower())
+            , (input_str.count(i) if i.isupper() else 0)
+            , input_str.lower().count(i.lower()) / len(self.remove_spaces(input_str)) * 100]
+               for i in set(self.remove_spaces(input_str))}
+        return res
+
+    def write_to_csv(self, file_to_read, headers, output_file, mode ):
+        with open(file_to_read, "r") as f:
+            contents = f.read()
+            if mode == "word_analysis":
+                contents_str = self.word_count(contents)
+            elif mode == "letter_analysis":
+                contents_str = self.analyze_count_letters(self.remove_delimiters(contents))
+            else:
+                print("Please specify word_analysis or letter_analysis mode")
+            with open(output_file, "w") as csvfile:
+                fieldnames = headers
+                writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+                writer.writeheader()
+                for k, v in contents_str.items():
+                    writer.writerow(dict(zip(headers, [k] + [v] if type(v)!= list else [k] + v)))
+
+class AnalyzeNews(PdqeNewsPaperSite):
+    def __init__(self,  output_file = "./newspaper.txt", input_file = "../module_6/test_message.txt"
+               , word_count_file_path = "./word_count.csv", analyze_text_csv = "./text_analysys.csv"):
+        super().__init__()
+        self.file_to_read = output_file
+        self.input_file_name = input_file
+        self.word_count_file_path = word_count_file_path
+        self.analyze_text_csv = analyze_text_csv
+
+        self.csv_transf = CsvTransformation()
+
+        sys.path.append(output_file)
+        sys.path.append(input_file)
+        sys.path.append(word_count_file_path)
+        sys.path.append(analyze_text_csv)
+
+        PdqeNewsPaperSite(self.file_to_read, self.input_file_name)
+        PdqeNewsPaperSite.user_input(self)
+        time.sleep(3)
+        self.csv_transf.write_to_csv(self.file_to_read, ['word', 'count'], self.word_count_file_path, "word_analysis")
+        self.csv_transf.write_to_csv(self.file_to_read, ["letter", "count", "uppercase_count", "percentage"]
+                                   , self.analyze_text_csv, "letter_analysis")
```

Comment:
In general, it is good, it is necessary to make minor corrections. 
According to the task criteria, it is necessary to count the number of words and letters.
But your code also counts empty characters and numbers. Try to fix it.
And also for beauty, I would round up the percentages and display, for example, 5.96%

Reply:
Thanks for highlighting this. Fixed